recipes/emsdk/all/conanfile.py

```diff
@@ -76,6 +72,13 @@ def generate(self):
         env.define_path("EM_CACHE", self._em_cache)
         env.vars(self, scope="emsdk").save_script("emsdk_env_file")
 
+        if cross_building(self):
```

Comment:
Strictly speaking, this doesn't need to be guarded in a conditional - quite the opposite, I'd say it's better to always set `EMSDK_ARCH` to match `self.settings.arch` - the architecture returned by the python interpreter may not always be `self.settings.arch` on systems that can run the interpreter on multiple architectures